scanpipe/pipes/purldb.py

```diff
@@ -99,24 +101,56 @@ def request_post(url, data, timeout=None):
         logger.debug(f"{label} [Exception] {exception}")
 
 
-def match_package(sha1, timeout=None, api_url=PURLDB_API_URL):
-    """Match a SHA1 in the PurlDB for package-type file."""
-    payload = {"sha1": sha1}
-    response = request_get(url=f"{api_url}packages/", payload=payload, timeout=timeout)
-
+def collect_response_results(response, data, timeout=None):
+    """Return all results from a purldb API response."""
+    results = []
     if response and response.get("count"):
-        results = response["results"]
-        return results
+        results.extend(response["results"])
+        next_page = response.get("next")
+        while next_page:
+            response = request_post(url=next_page, data=data, timeout=timeout)
+            if response and response.get("count"):
+                results.extend(response["results"])
+            next_page = response.get("next")
+    return results
+
+
+def match_packages(sha1_list, timeout=None, api_url=PURLDB_API_URL):
+    """Match a list of SHA1 in the PurlDB for package-type files."""
+    data = {"sha1": sha1_list}
```

Comment:
@JonoYang please add `enhance_package_data` as well in this PR itself, if it makes sense

Reply:
Ack, I've made the update.